ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/request/AccountRangeDataRequest.java

```diff
@@ -118,7 +124,15 @@ protected int doPersist(
           nbNodesSaved.getAndIncrement();
         };
 
-    stackTrie.commit(nodeUpdater);
+    StackTrie.FlatDatabaseUpdater flatDatabaseUpdater = noop();
+    if (worldStateStorage.getDataStorageFormat().equals(DataStorageFormat.BONSAI)) {
+      // we have a flat DB only with Bonsai
+      flatDatabaseUpdater =
+          (key, value) ->
+              ((BonsaiWorldStateKeyValueStorage.BonsaiUpdater) updater)
+                  .putAccountInfoState(Hash.wrap(key), value);
+    }
```

Comment:
seems like these flatDatabaseUpdaters could be provided from worldStateStorage that way we don't have storage format conditional logic embedded into range requests (if/when we implement another storage format)

Reply:
not fan to do that it's like returning nodeupdater on worldstatestorage . maybe we need to find another way or maybe having worldstate storage more bonsai friendly and less forest friendly

Reply:
I modified the code to have something more generic